authentication/views.py

```diff
@@ -232,3 +240,50 @@ def set_password(self, request, *args, **kwargs):
             update_session_auth_hash(self.request, self.request.user)
             return Response({}, status=status.HTTP_200_OK)
         return response
+
+
+@api_view(["GET", "POST"])
+@permission_classes([IsAuthenticated, IsStaffPermission])
+@authentication_classes([BearerAuthentication])
+def get_user_details_for_keycloak(request, email):
+    """
+    Endpoint for the Keycloak plug-in: https://github.com/daniel-frak/keycloak-user-migration.
+
+    Args:
+        email (string): The email of a user record in Open Discussions.
```

Comment:
Yeah this will only support email coming from Keycloak.  The reasoning being is that each SSO client (Open Discussions, MIT Open, MITx Online) we intend to connect with this plug-in could have user's with the same username.  We don't have any way to verify that they are indeed the same user.  

So I think our best course of action is to keep the management of usernames to within their respective client applications, and just set the user's Keycloak username to their email.